### PR TITLE
add link to CuPy upgrade guide

### PR DESCRIPTION
**Pull request by kmaehashi**
Add link to CuPy upgrade guide from Chaier upgrade guide.

~~Depends on https://github.com/cupy/cupy/pull/884.~~ (merged)

### CODE REVIEW

File: docs/source/upgrade.rst [hunk 1]
```diff
@@ -59,6 +59,13 @@ This change was introduced because CUDA 7.5 does not support NVIDIA Pascal GPUs.
 To use these images, you may need to upgrade the NVIDIA driver on your host.
 See `Requirements of nvidia-docker <https://github.com/NVIDIA/nvidia-docker/wiki/CUDA#requirements>`_ for details.
 
+CuPy v4
+-------
+
+Chainer v4 requires CuPy v4 if you need GPU support.
+Please see the `Upgrade Guide for CuPy v4 <https://docs.chainer.org/en/latest/upgrade.html#cupy-v4>`_ for details.
```
[the remark this answers is not in the document]
Comment (kmaehashi) [reply docs/source/upgrade.rst]: Thanks, that was wrong. Fixed.

**Comment by delta2323** [comment]
jenkins, test this please.

**Comment by delta2323** [comment]
Thank you for update. I'm thinking to merge this after cupy/cupy#884.

**Comment by delta2323** [comment]
LGTM

**Pull request merged**